angular-chart.js

```diff
@@ -238,15 +241,53 @@
     }
 
     function getColor (color) {
-      return {
-        backgroundColor: rgba(color, 0.2),
-        pointBackgroundColor: rgba(color, 1),
-        pointHoverBackgroundColor: rgba(color, 0.8),
-        borderColor: rgba(color, 1),
-        pointBorderColor: '#fff',
-        pointHoverBorderColor: rgba(color, 1)
+      function colorSetter(colorArray, aVal){
+        return {
+          backgroundColor: rgba(colorArray, 0.2),
+          pointBackgroundColor: rgba(colorArray, aVal),
+          pointHoverBackgroundColor: rgba(colorArray, 0.8),
+          borderColor: rgba(colorArray, aVal),
+          pointBorderColor: '#fff',
+          pointHoverBorderColor: rgba(colorArray, aVal)
+        };
       };
-    }
+      // If the color is a Hex color, this processes the array that hexToRgb() returns.
+      if (typeof color === 'object'){
+        return colorSetter(color, 1);
+      }
+      if (typeof color === 'string'){
+        // If the color is an RGB/RGBA color, this finds the indexes of all commas in the string, then converts
+        // the RGBA or RGB string into an array for processing by the rgba() function when it tests the browser for RGBA color support.
+        var rgbaArray=[];
+        var commaIndexes = [];
+        var position = color.indexOf(',');
+        while (position !== -1) {
+          commaIndexes.push(position);
+          position = color.indexOf(',', position + 1);
+        };
+        // Converts the first RGB/RGBA number in the string into a number and pushes it into the array.
+        if (commaIndexes.length === 2) {
+          rgbaArray.push(color.substring(4, commaIndexes[0]));
+        } else {
+          rgbaArray.push(color.substring(5, commaIndexes[0]));
+        };
+        // Converts the second RGB/RGBA numbers in the string into numbers and pushes them into the array
+        rgbaArray.push(color.substring(commaIndexes[0]+1, commaIndexes[1]));
+        // Converts the third RGB/RGBA number in the string into a number and pushes it into the array.          
+        if (commaIndexes.length === 2){
+          rgbaArray.push(color.substring(commaIndexes[1]+1, color.length-1)); 
+        // Sets the A value for an RGB color to 1.
+          rgbaArray.push(1);        
+        } else {
+          rgbaArray.push(color.substring(commaIndexes[1]+1, commaIndexes[2]));
+        // If RGBA number is used, converts the fourth RGBA number in the string into a number and pushes it into the array.                      
+          rgbaArray.push(color.substring(commaIndexes[2]+1, color.length-1));  
+        };         
+        color = rgbaArray.slice(0,3);
+        var aVal = rgbaArray[3];
```

Comment:
Instead of testing the length etc. you can just used the fact that `rgbaArray[3]` will return `undefined` if it doesn't exist and set it to 1 instead so basically the entire logic can be changed to:

```
function getColor (color) {
   color = color.slice(0, 3);
   var alpha = color[3] || 1;
   return { ... };
}

function rgbStringToRgb (color) {
    return color.split(',');
}
```